### PR TITLE
API Implemented the non-deprecated interface for credential repository

### LINKED ISSUE #11: Replace use of deprecated Webauthn\CredentialRepository interface
The `Webauthn\CredentialRepository` interface is deprecated in favour of `Webauthn\PublicKeyCredentialSourceRepository`. We should update it in `CredentialRepository`

See https://github.com/web-auth/webauthn-framework/blob/v1.2/doc/webauthn/PublicKeyCredentialCreation.md#credential-repository-public-key-credential-source-repository

### PR DESCRIPTION
**Pull request by ScopeyNZ**
This module now supports multiple registration for WebAuthn although it is not possible through the UI. The repository itself is stored against the registered method.

Resolves #11

### CODE REVIEW

File: src/CredentialRepository.php [hunk 1]
```diff
@@ -3,99 +3,213 @@
...
+    }
+
+    /**
+     * Constructs the object
+     * @link https://php.net/manual/en/serializable.unserialize.php
+     * @param string $serialized <p>
+     * The string representation of the object.
+     * </p>
+     * @return void
+     * @since 5.1.0
```
Comment (robbieaverill) [inline src/CredentialRepository.php]: These doc blocks seem to be from PHP internals - maybe remove? They should inherit from the interface if you don't define them

File: src/VerifyHandler.php [hunk 2]
```diff
@@ -132,7 +136,7 @@ public function getComponent(): string
      */
     protected function getCredentialRequestOptions(
         StoreInterface $store,
-        RegisteredMethod $registeredMethod,
+        RegisteredMethod $registeredMethod = null,
```
Comment (robbieaverill) [inline src/VerifyHandler.php]: Doc block could use a tweak for being nullable and the new AuthenticationFailedException

**Comment by robbieaverill** [comment]
Update branch to master, since we're in RC?

**Comment by brynwhyman** [comment]
Resolves #11

**Comment by ScopeyNZ** [comment]
We've spoken a little about this PR "off line". It's a risk having a PR like this so close to RC.

Risks:
- Close to RC - could introduce bugs
- Additional non-audited code

Rebuttal:
- RC phases are usually to indicate the software is feature complete, but (potentially breaking) bug fixes are still ongoing
- We already have plenty of non-audited code _and_ this change is implementing the existing 3rd party library in their recommended way - that library has not been updated

Payoffs:
- We will have to wipe existing authenticators if we do this in the future so we're preventing that
- Might be able to get away with multiple authenticators in a future minor release rather than major.

**Pull request merged**